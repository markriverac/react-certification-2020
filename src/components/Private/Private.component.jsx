import React, { useContext } from 'react';
import { Route, Redirect } from 'react-router-dom';
import UserContext from '../../providers/UserProvider/User.context';

function Private({ children, ...rest }) {
  const user = useContext(UserContext);
  return (
    <Route
      {...rest}
      render={() => (user.userState.isAuth ? children : <Redirect to="/" />)}
    />
  );
}

export default Private;
