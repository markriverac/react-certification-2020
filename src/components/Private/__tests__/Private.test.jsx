import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';
import Private from '../Private.component';
import UserContext from '../../../providers/UserProvider/User.context';

Enzyme.configure({ adapter: new Adapter() });

it('Renders without exploding', () => {
  const userState = { isAuth: false };
  const notFound = renderer.create(
    <BrowserRouter>
      <UserContext.Provider value={{ userState }}>
        <Private />
      </UserContext.Provider>
    </BrowserRouter>
  );
  const tree = notFound.toJSON();
  expect(tree).toMatchSnapshot();
});
