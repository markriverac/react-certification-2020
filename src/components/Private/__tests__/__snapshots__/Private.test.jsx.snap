// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without exploding 1`] = `null`;
